src/components/auth-single-form.ts

```diff
@@ -0,0 +1,129 @@
+import '@kano/styles/typography.js';
+import { button } from '@kano/styles/button.js';
+import { LitElement, html, property, query } from 'lit-element/lit-element.js';
+import { templateContent } from '../utils/template-content.js';
+import { styles } from '../styles.js';
+import { validateUsername, validatePassword, validateEmail } from '../utils/validation.js';
+
+interface SubmitDetails {
+    detail: EventDetail;
+}
+
+interface EventDetail {
+    next: string;
+    payload: EventPayload;
+}
+
+interface EventPayload {
+    [key: string]: string;
+}
+interface Errors {
+    [key: string]: string;
+}
+
+
+export class SingleInputElement extends LitElement {
+    static get styles() {
+            return [styles];
+        } 
+    @property({ type: Object }) errors : Errors = {};
+    @property({ type: String }) id = 'username';
+    @property({ type: String }) next = 'password';
+
+    @query('#input')
+    public inputElement? : HTMLInputElement;
+   
+    /**
+     * Returns the current value of the username field or an empty string
+     */
+    get value() {
+        return this.inputElement ? this.inputElement.value : '';
+    }
+
+    inputTemplate() {
+        // replace with your specific template
+        return html`
+            <div class="input-wrapper">
+                <input
+                    @blur="${() => this.validateInput(this.id, this.value)}"
+                    type="text"
+                    id="input"
+                    placeholder="Input information here"/>
+                <div class="error">${this.errors[this.id]}</div>
+            </div>
+        `;
+    }
+
+    render() {        
+        return html`
+        ${templateContent(button)}
+        <div class="form-template">
+            <div>
+                ${this.inputTemplate()}
+                <div class="button-wrapper">
+                    <button
+                    id="submit"
+                    @mousedown=${(e: Event) => this.handleClick(e)}
```

Comment:
This surely is not gonna work on touch devices

Reply:
I looked it up and seems that touch events emulate mouse events:
https://developer.mozilla.org/en-US/docs/Web/API/Touch_events/Supporting_both_TouchEvent_and_MouseEvent